sklearn/tree/_splitter.pxd

```diff
@@ -59,6 +107,8 @@ cdef class BaseSplitter:
 
     cdef const float64_t[:] sample_weight
 
+    cdef SplitRecordFactoryClosure split_record_factory
```

Comment:
Why is it named Closure?

Reply:
Because it is a cython implementation of a closure. C doesn't support closures as a language level feature, but a struct of a function pointer bound with a struct of variable values functions the same.